torchmultimodal/modules/losses/albef.py

```diff
@@ -0,0 +1,216 @@
+        )
+        vl_output = self.itm_head(vl_embeddings)
+        itm_labels = torch.cat(
+            [
+                torch.ones(batch_size, dtype=torch.long),
+                torch.zeros(2 * batch_size, dtype=torch.long),
+            ],
+            dim=0,
+        ).to(vl_embeddings.device)
+        loss_itm = F.cross_entropy(vl_output, itm_labels)
+        return loss_itm
+
+
+class MaskedLanguageModelingLoss(nn.Module):
+    """
+    Compute the autoregressive masked language modeling loss by predicting the next token.
+
+    Args:
+        masked_token_id (int): The token id indicating a masked token. Default is -100.
+        vocab_size (int): The number of different tokens the prediction_head can predict. Default is 30522.
+        hidden_size (int): The hidden size of the prediction_head. Default is 768.
+        layer_norm_eps (float): The epsilon used by the prediction_head normalization layer. Default is 1e-12.
+        alpha (float): The interpolation value between mlm_loss and loss_distill. Default is 0.
+        transform_act_fn (Callable[[Tensor], Tensor]): The activation function in the prediction_head. Default is GELU.
+
+    Inputs:
+        labels (Tensor): The masked output tokens.
+        hidden_states (Tensor): The hidden states of preceding tokens.
+        hidden_states_m (Tensor): The hidden states of preceding tokens from momentum models.
+        weights (Tensor): The weight for each output.
+    """
+
+    def __init__(
+        self,
+        mask_token_id: int = -100,
+        vocab_size: int = 30522,
+        hidden_size: int = 768,
+        layer_norm_eps: float = 1e-12,
+        alpha: float = 0.0,
+        transform_act_fn: Callable[[Tensor], Tensor] = nn.functional.gelu,
+    ) -> None:
+        super().__init__()
+        self.mask_token_id = mask_token_id
+        self.vocab_size = vocab_size
+        self.alpha = alpha
+        self.prediction_head = PredictionHead(
+            vocab_size, hidden_size, layer_norm_eps, transform_act_fn
+        )
+        self.loss_fn = CrossEntropyLoss(reduction="none")
+
+    def forward(
+        self,
+        labels: Tensor,
+        hidden_states: Tensor,
+        hidden_states_m: Tensor,
+        weights: Tensor,
+    ) -> Tensor:
+        batch_size = labels.size(0)
+        prediction_scores = self.prediction_head(hidden_states)
+        # shift prediction scores and labels by one for next-token prediction
+        prediction_scores = prediction_scores[:, :-1, :].contiguous()
+        labels = labels[:, 1:].contiguous()
+        with torch.no_grad():
+            prediction_scores_m = self.prediction_head(hidden_states_m)
+            prediction_scores_m = prediction_scores_m[:, :-1, :].contiguous()
+        loss_distill = -torch.sum(
+            F.log_softmax(prediction_scores, dim=-1)
+            * F.softmax(prediction_scores_m, dim=-1),
+            dim=-1,
+        )
+        loss_distill = (loss_distill * (labels != self.mask_token_id)).sum(1)
```

Comment:
If alpha = 0.0, we don't need to compute this, right? And also `hidden_states_m` becomes optional. Might be a nice way to make this work well with a more standard kind of MLM loss. (Though it will make the class a bit more convoluted, so I'll leave it up to you if you wanna do this or not)